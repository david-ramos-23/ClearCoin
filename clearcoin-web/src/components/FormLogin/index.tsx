import React from 'react';
import styled from 'styled-components';
import { useForm } from 'react-hook-form';

const FormContainer = styled.div`
	display: flex;
	flex-flow: row wrap;
	margin: 0;
	padding: 0;
	justify-content: center;
	margin: 40px 20px 20px;
`;

const Form = styled.form`
	display: flex;
	flex-direction: column;
	width: 300px;
`;

const StyledInput = styled.input`
	font-size: 16px;
	height: 38px;
	background-color: #eefbf7;
	padding: 0 .5rem;
	color: #2d3134;
	margin: 5px;
	border: 1px solid #bfc5ca;
	border-radius: 4px;
	font-weight: 200;
	line-height: 1.4;

	&:focus {
		border-color: #b2fce4;
		box-shadow: 0 1px 1px #000,inset .75,0 0 8px #eaf5f7,.6;
		outline: 0 none;
	}
`;

const Button = styled.button`

	margin: 5px;
	background: #b2fce4;
	border: .125rem solid transparent;
	border-radius: .75rem;
	box-shadow: none;
	color: #000;
	font-weight: 400;
	transition: all .5s ease;
	font-size: 1.125rem;
	padding: .75rem 1.125rem;
	
	&:hover {
		cursor: pointer;
		background-color: #828282;
		color: #fff;
	}
`;

type FormLoginType = {
	onSubmitForm: Function;
}

const FormLogin = ({ onSubmitForm }: FormLoginType) => {

	const { register, handleSubmit } = useForm();
	const onSubmit = (data: any) => {
		onSubmitForm(data);
	};

	return (
		<FormContainer>
			<Form onSubmit={handleSubmit(onSubmit)}>
				<StyledInput required {...register('email')} placeholder="email" type="email"/>
				<StyledInput
					required
					{...register('password')}
					type="password"
					className="password"
					placeholder="password"
				/>
				<Button type="submit">
					Submit
				</Button> 
			</Form>
		</FormContainer>
	);
};

export default FormLogin;
