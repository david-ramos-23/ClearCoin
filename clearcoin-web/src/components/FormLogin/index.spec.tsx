import React from 'react';
import { render, screen } from '@testing-library/react';
import FormLogin from './index';

describe('FormLogin', () => {
	test('Render FormLogin', () => {
		render(<FormLogin onSubmitForm={() => console.log('submitted')}/>);

		expect(screen.getByPlaceholderText('email')).toBeInTheDocument();
		expect(screen.getByPlaceholderText('password')).toBeInTheDocument();
		expect(screen.getByText('Submit')).toBeInTheDocument();
	});

	test('snapshot is ok', () => {
		const { container } = render(<FormLogin onSubmitForm={() => console.log('submitted')}/>);

		expect(container).toMatchSnapshot();
	});
});
