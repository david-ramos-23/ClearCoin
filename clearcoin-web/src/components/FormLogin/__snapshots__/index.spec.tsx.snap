// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLogin snapshot is ok 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin: 0;
  padding: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 40px 20px 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 300px;
}

.c2 {
  font-size: 16px;
  height: 38px;
  background-color: #eefbf7;
  padding: 0 .5rem;
  color: #2d3134;
  margin: 5px;
  border: 1px solid #bfc5ca;
  border-radius: 4px;
  font-weight: 200;
  line-height: 1.4;
}

.c2:focus {
  border-color: #b2fce4;
  box-shadow: 0 1px 1px #000,inset .75,0 0 8px #eaf5f7,.6;
  outline: 0 none;
}

.c3 {
  margin: 5px;
  background: #b2fce4;
  border: .125rem solid transparent;
  border-radius: .75rem;
  box-shadow: none;
  color: #000;
  font-weight: 400;
  -webkit-transition: all .5s ease;
  transition: all .5s ease;
  font-size: 1.125rem;
  padding: .75rem 1.125rem;
}

.c3:hover {
  cursor: pointer;
  background-color: #828282;
  color: #fff;
}

<div>
  <div
    class="c0"
  >
    <form
      class="c1"
    >
      <input
        class="c2"
        name="email"
        placeholder="email"
        required=""
        type="email"
      />
      <input
        class="c2 password"
        name="password"
        placeholder="password"
        required=""
        type="password"
      />
      <button
        class="c3"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
